src/main/java/org/jenkins/tools/test/model/PluginRemoting.java

```diff
@@ -106,181 +93,34 @@ private String retrievePomContentFromXmlFile() throws IOException {
-            LOGGER.log(Level.WARNING, "Failed to retrieve SCM connection", e);
-            throw new PluginSourcesUnavailableException("Failed to retrieve SCM connection", e);
-        }
-
-        PomData pomData =
-                new PomData(artifactId, packaging, scmConnection, scmTag, parent, groupId);
-        computeScmConnection(pomData);
-        return pomData;
-    }
-
-    /**
-     * Retrieves a field value by XPath. The value must exist and be non-empty.
-     *
-     * @throws IOException parsing error
-     */
-    @NonNull
-    private static String getValueOrFail(Document doc, XPath xpath, String field)
-            throws IOException {
-        String res;
-        try {
-            res = xpath.evaluate(field + "/text()", doc);
-        } catch (XPathExpressionException e) {
-            throw new IOException("Expression failed for the field " + field, e);
-        }
-
-        if (res == null || res.isBlank()) {
-            throw new IOException("Field is either null or blank: " + field);
-        }
-        return res;
-    }
-
-    public static void computeScmConnection(PomData pomData) {
-        String transformedConnectionUrl = pomData.getConnectionUrl();
-
-        // Trimming url
-        transformedConnectionUrl = transformedConnectionUrl.trim();
-
-        // Generally, when connectionUrl is empty, is implies it is declared in a parent pom
-        // => Only possibility is to deduct github repository from artifactId (crossing fingers it
-        // is not a bizarre repository url...)
-        String oldUrl = transformedConnectionUrl;
-        if (transformedConnectionUrl.isEmpty()) {
-            transformedConnectionUrl =
-                    "scm:git:git://github.com/jenkinsci/"
-                            + pomData.artifactId.replaceAll("jenkins", "")
-                            + "-plugin.git";
-            if (!oldUrl.equals(transformedConnectionUrl)) {
-                LOGGER.log(
-                        Level.WARNING,
-                        "project.scm.connectionUrl is not present in plugin's pom .. isn't it"
-                                + " residing somewhere on a parent pom ?");
-            }
-        }
-
-        // Java.net SVN migration
-        oldUrl = transformedConnectionUrl;
-        transformedConnectionUrl =
-                transformedConnectionUrl.replaceAll(
-                        "(svn|hudson)\\.dev\\.java\\.net/svn/hudson/",
-                        "svn.java.net/svn/hudson~svn/");
-        if (!oldUrl.equals(transformedConnectionUrl)) {
-            LOGGER.log(
-                    Level.WARNING,
-                    "project.scm.connectionUrl is pointing to svn.dev.java.net/svn/hudson/ instead"
-                            + " of svn.java.net/svn/hudson~svn/");
-        }
-
-        // ${project.artifactId}
-        transformedConnectionUrl =
-                transformedConnectionUrl.replaceAll(
-                        "\\$\\{project\\.artifactId\\}", pomData.artifactId);
```

Comment:
I haven't seen a POM like this in years and Maven 4 actually won't accept it.

Reply:
for the repository and pluginsRepository section - not for scm?  (not that it matters here - but if you we are no longer able to use properties in an SCM connection there are a lot of plugins that need to be fixed for maven 4.

Reply:
I saw it at least for `dependencies.dependency.groupId` in [JENKINS-70478](https://issues.jenkins.io/browse/JENKINS-70478).